### PR TITLE
Notify change in balance when updating network fee

### PR DESCRIPTION
**Pull request by pm47**
As a funder, when we send an `update_fee`, the channel fails to notify
other parts of the code (e.g. the relayer), causing "insufficient funds"
errors down the road.

### CODE REVIEW

**Comment by pm47** [comment]
@t-bast not sure what's wrong with the integration tests

File: eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala [hunk 1]
```diff
@@ -751,6 +749,10 @@ class Channel(val nodeParams: NodeParams, val wallet: EclairWallet, remoteNodeId
             // if we have newly acknowledged changes let's sign them
             self ! CMD_SIGN
           }
+          if (commitments1.availableBalanceForSend != commitments1.availableBalanceForSend) {
```
Comment (t-bast) [inline eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala]:
```suggestion
          if (d.commitments.availableBalanceForSend != commitments1.availableBalanceForSend) {
```

File: eclair-core/src/main/scala/fr/acinq/eclair/channel/ChannelEvents.scala [hunk 2]
```diff
@@ -53,7 +53,9 @@ case class ChannelErrorOccurred(channel: ActorRef, channelId: ByteVector32, remo
 case class NetworkFeePaid(channel: ActorRef, remoteNodeId: PublicKey, channelId: ByteVector32, tx: Transaction, fee: Satoshi, txType: String) extends ChannelEvent
 
 // NB: this event is only sent when the channel is available
-case class AvailableBalanceChanged(channel: ActorRef, channelId: ByteVector32, shortChannelId: ShortChannelId, localBalance: MilliSatoshi, commitments: Commitments) extends ChannelEvent
+case class AvailableBalanceChanged(channel: ActorRef, channelId: ByteVector32, shortChannelId: ShortChannelId, commitments: Commitments) extends ChannelEvent {
+  def localBalance: MilliSatoshi = commitments.remoteNextCommitInfo.left.toOption.map(_.nextRemoteCommit).getOrElse(commitments.remoteCommit).spec.toRemote
```
Comment (t-bast) [inline eclair-core/src/main/scala/fr/acinq/eclair/channel/ChannelEvents.scala]: I would completely remove this `localBalance` field and the corresponding DB table (which is probably very noisy).

**Comment by t-bast** [comment]
We probably need to add this check in all outgoing update_* commands...

**Pull request merged**